Geometry/CaloGeometry/src/FlatHexagon.cc

```diff
@@ -96,6 +96,18 @@ GlobalPoint FlatHexagon::getPosition(const Pt3D& local) const {
   return GlobalPoint(glb.x(), glb.y(), glb.z());
 }
 
+float FlatHexagon::etaSpan() const {
+  float eta1 = -std::log(std::tan(0.5 * std::atan((m_global.perp() + param()[k_R]) / std::abs(m_global.z()))));
```

Comment:
is it possible to cache these values (and the ones in phiSpan below) so the calculation doesn't have to be repeated every time the accessor is called?

Reply:
Thread safe (and efficient) caching is a tricky business.

Reply:
This is a pretty simple class, though. the dEta and dPhi values can just be calculated and stored whenever m_global is changed (only in constructors or setPosition() accessor)

Reply:
It is used in HGCalGeometry and m_global is changed when the CaloCellGeometry pointer is asked. In most of the geometry applications, this function is not required. It should be heavy to calculate this every time it is loaded. Most likely this is required in debugging or in some analysis code. It is better to calculate them only when it is needed.